app/assets/javascripts/discourse/lib/dialog-holder/addon/components/dialog-holder.hbs

```diff
@@ -0,0 +1,35 @@
+<div id="dialog-holder" class="dialog-container {{this.dialog.classes}}" aria-labelledby={{this.dialog.titleElementId}} aria-hidden="true">
+  <div class="dialog-overlay" data-a11y-dialog-hide></div>
+
+  {{#if this.dialog.type}}
+    <div class="dialog-content" role="document">
+      {{#if this.dialog.title}}
+        <div class="dialog-header">
+          <h3 id="dialog-title">{{this.dialog.title}}</h3>
+          <DButton @icon="times" @action={{action this.dialog.cancel}} @class="btn-flat dialog-close close" @title="modal.close" />
+        </div>
+      {{/if}}
+
+      {{#if this.dialog.message}}
+        <div class="dialog-body">
+          {{html-safe this.dialog.message}}
+        </div>
+      {{/if}}
+
+      {{#if (notEq this.dialog.type "notice")}}
+        <div class="dialog-footer">
+          {{#if this.dialog.buttons}}
+            {{#each this.dialog.buttons as |button|}}
+              <DButton @icon={{button.icon}} @class={{button.class}} @action={{action "buttonAction" button}} @translatedLabel={{button.label}} />
+            {{/each}}
+          {{else}}
+            <DButton @class="btn-primary" @action={{this.dialog.didConfirmWrapped}} @icon={{this.dialog.iconConfirm}} @label={{this.dialog.labelConfirm}} />
+            {{#if this.dialog.cancelVisible}}
```

Comment:
Im not sure what should be our convention here, but maybe:

- shouldDisplayCancel
- isCancelVisible

reads better IMO